### PR TITLE
Docs: Add missing documentation for enterprise features

### PR DESCRIPTION
**Pull request by mmandrus**
**What this PR does / why we need it**:
This PR adds documentation for a few enterprise features that had little-to-no mention in the gcom docs:
- User protection (add mention of it to top-level enterprise pages and expand on it on security config page)
- LDAP Sync Debug (add to existing LDAP admin page section)
- Recorded queries (add mention to top-level enterprise pages)

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/3932

**Special notes for your reviewer**:

Note that the LDAP Sync Debug comes with a screenshot, which has already been uploaded to the website: https://grafana.com/static/img/docs/ldap_sync_debug.png

Backport was recommended for these changes.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/ldap.md [hunk 1]
```diff
@@ -120,6 +120,10 @@ To use the debug view:
 
 {{< figure src="/static/img/docs/ldap_debug_mapping_testing.png" class="docs-image--no-shadow" max-width="600px" >}}
 
+[Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise/" >}}) users with [enhanced LDAP integration]({{< relref "./enhanced_ldap/" >}}) enabled can also see sync status in the debug view. This requires the `ldap.status:read` permission.
```
Comment (Eve832) [inline docs/sources/setup-grafana/configure-security/configure-authentication/ldap.md:123]:
```suggestion
[Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise/" >}}) users with [enhanced LDAP integration]({{< relref "enhanced_ldap" >}}) enabled can also see sync status in the debug view. This requires the `ldap.status:read` permission.
```
It looks like this second relref points to a file in the same folder, so you can just call it like this

**Review by Eve832 — CHANGES_REQUESTED** [review]
Thank you @mmandrus !!!!

File: docs/sources/enterprise/_index.md [hunk 2]
```diff
@@ -65,6 +69,7 @@ Grafana Enterprise adds the following features:
 - [Export dashboard as PDF]({{< relref "../dashboards/share-dashboards-panels/#export-dashboard-as-pdf" >}})
 - [Custom branding]({{< relref "../setup-grafana/configure-grafana/configure-custom-branding/" >}}) to customize Grafana from the brand and logo to the footer links.
 - [Usage insights]({{< relref "../dashboards/assess-dashboard-usage/" >}}) to understand how your Grafana instance is used.
+- [Recorded queries]({{< relref "./recorded-queries.md" >}}) to see trends over time for your datasources.
```
[the remark this answers is not in the document]
Comment (mmandrus) [reply docs/sources/enterprise/_index.md:72]: It is actually in the same directory -- I just updated it to follow the other examples in the file. Thanks!

File: docs/sources/enterprise/_index.md [hunk 3]
```diff
@@ -65,6 +69,7 @@ Grafana Enterprise adds the following features:
 - [Export dashboard as PDF]({{< relref "../dashboards/share-dashboards-panels/#export-dashboard-as-pdf" >}})
 - [Custom branding]({{< relref "../setup-grafana/configure-grafana/configure-custom-branding/" >}}) to customize Grafana from the brand and logo to the footer links.
 - [Usage insights]({{< relref "../dashboards/assess-dashboard-usage/" >}}) to understand how your Grafana instance is used.
+- [Recorded queries]({{< relref "recorded-queries/" >}}) to see trends over time for your datasources.
```
Comment (Eve832) [inline docs/sources/enterprise/_index.md:72]:
```suggestion
- [Recorded queries]({{< relref "recorded-queries" >}}) to see trends over time for your data sources.
```

File: docs/sources/introduction/grafana-enterprise.md [hunk 4]
```diff
@@ -56,6 +60,7 @@ Grafana Enterprise adds the following features:
 - [Export dashboard as PDF]({{< relref "../dashboards/share-dashboards-panels/#export-dashboard-as-pdf" >}})
 - [Custom branding]({{< relref "../setup-grafana/configure-grafana/configure-custom-branding/" >}}) to customize Grafana from the brand and logo to the footer links.
 - [Usage insights]({{< relref "../dashboards/assess-dashboard-usage/" >}}) to understand how your Grafana instance is used.
+- [Recorded queries]({{< relref "../enterprise/recorded-queries/" >}}) to see trends over time for your datasources.
```
Comment (Eve832) [inline docs/sources/introduction/grafana-enterprise.md:63]:
```suggestion
- [Recorded queries]({{< relref "../enterprise/recorded-queries" >}}) to see trends over time for your data sources.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/grafana.md [hunk 5]
```diff
@@ -123,3 +123,18 @@ URL to redirect the user to after signing out from Grafana. This can for example
 [auth]
 signout_redirect_url =
 ```
+
+### Protected roles
+
+> **Note:** Available in [Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise/" >}}) and [Grafana Cloud Advanced]({{< ref "../../../introduction/grafana-cloud" >}}).
```
Comment (Eve832) [inline docs/sources/setup-grafana/configure-security/configure-authentication/grafana.md:129]:
```suggestion
> **Note:** Available in [Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise" >}}) and [Grafana Cloud Advanced]({{< relref "../../../introduction/grafana-cloud" >}}).
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/ldap.md [hunk 6]
```diff
@@ -120,6 +120,10 @@ To use the debug view:
 
 {{< figure src="/static/img/docs/ldap_debug_mapping_testing.png" class="docs-image--no-shadow" max-width="600px" >}}
 
+[Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise/" >}}) users with [enhanced LDAP integration]({{< relref "enhanced_ldap" >}}) enabled can also see sync status in the debug view. This requires the `ldap.status:read` permission.
```
Comment (Eve832) [inline docs/sources/setup-grafana/configure-security/configure-authentication/ldap.md:123]:
```suggestion
[Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise" >}}) users with [enhanced LDAP integration]({{< relref "enhanced_ldap" >}}) enabled can also see sync status in the debug view. This requires the `ldap.status:read` permission.
```

**Review by Eve832 — CHANGES_REQUESTED** [review]

**Review by Eve832 — APPROVED** [review]

**Pull request merged**